massa-bootstrap/src/client.rs

```diff
@@ -437,15 +437,71 @@ pub async fn get_state(
         }
         return Ok(GlobalBootstrapState::new(final_state));
     }
+
+    let mut filtered_bootstrap_list = bootstrap_config.bootstrap_list.clone();
+
+    // We filter the bootstrap list to keep only the ip addresses we are compatible with
+    match &bootstrap_config.bootstrap_protocol {
+        Some(s) if s.as_str() == "ipv4" => {
+            let prev_bootstrap_list_len = filtered_bootstrap_list.len();
+
+            filtered_bootstrap_list = filtered_bootstrap_list
+                .clone()
+                .into_iter()
+                .filter(|&(addr, _)| addr.is_ipv4())
+                .collect();
+
+            let new_bootstrap_list_len = filtered_bootstrap_list.len();
+
+            debug!(
+                "Filtered out {} IPv6 bootstrap addresses out of a total of {} bootstrap servers.",
+                prev_bootstrap_list_len as i32 - new_bootstrap_list_len as i32,
+                prev_bootstrap_list_len
+            );
+        }
+        Some(s) if s.as_str() == "ipv6" => {
+            let prev_bootstrap_list_len = filtered_bootstrap_list.len();
+
+            filtered_bootstrap_list = filtered_bootstrap_list
+                .clone()
+                .into_iter()
+                .filter(|&(addr, _)| addr.is_ipv6())
+                .collect();
+
+            let new_bootstrap_list_len = filtered_bootstrap_list.len();
+
+            debug!(
+                "Filtered out {} IPv4 bootstrap addresses out of a total of {} bootstrap servers.",
+                prev_bootstrap_list_len as i32 - new_bootstrap_list_len as i32,
+                prev_bootstrap_list_len
+            );
+        }
+        _ => {}
```

Comment:
May be, we should be explicit here. "ipv4", "ipv6", "both" or thow an error

Reply:
Should I create a specific enum for it? So that if it is invalid it produces the error when it tries to deserialize the config instead.

Reply:
It seems to fit in enum, so i would say yes